### PR TITLE
Adds support for arguments to `get_regional_data`

### PR DESCRIPTION
**Pull request by seabbs**
This PR adds support for adding arguments to `get_regional_data`. This is required so that @kathsherratt can add new deaths, and admissions data sources for the UK (gov request + adding to website at a later date). Another PR will follow adding support for these datasets (which need to be processed with top priority after UK cases).

### CODE REVIEW

File: R/entities.R [hunk 1]
```diff
@@ -45,13 +45,15 @@ Region <- R6Class("Region",
...
                                 cases_subregion_source = "region_level_1",
+                                data_args = NULL,
                                 initialize = function(name,
                                                       covid_regional_data_identifier = NA,
                                                       case_modifier = NA,
                                                       generation_time = NA,
                                                       incubation_period = NA,
                                                       reporting_delay = NA,
                                                       cases_subregion_source = "region_level_1",
+                                                      data_args = NULL,
```
Comment (joeHickson) [inline R/entities.R:56]: you need to store this value (e.g. ln 66) as sadly R6 won't automatically map a param onto the class variable of the same name.

**Review by joeHickson — CHANGES_REQUESTED** [review]
Store the value for later use, otherwise good.

File: R/entities.R (hunk 1 above)
Comment (seabbs) [reply R/entities.R:56]: now done

**Review by joeHickson — APPROVED** [review]
LGTM

**Pull request merged**